src/code-util.jl

```diff
@@ -123,25 +123,55 @@ function is_special_symbol(symbol::Symbol)::Bool
     return symbol == :abort || symbol == :output || symbol == :rel
 end
 
-# Generate a string representing the Rel type for the input
-# Expected inputs are a vector of types, a tuple of types, or a type
-function type_string(input::Vector)
-    if isempty(input)
-        return ""
+# Generate a string representing the Rel type for input Vectors
+# :a => [...]
+function type_string(input::Vector{T}) where T
```

Comment:
is it possible for this to be a `Union` type? I think with the way we do specializations, the answer is no - is that your understanding as well?

Reply:
I wasn't sure what you were thinking of Union-ing, but it inspired me to simplify the methods.